src/EditorFeatures/CSharpTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.cs

```diff
@@ -0,0 +1,276 @@
+        public void TestUpdateGetSet_ExpressionBodies()
+        {
+            Test(
+@"using System; class C { int [||]GetFoo() => 0; void SetFoo(int i) => Bar(); }",
+@"using System; class C { int Foo { get { return 0; } set { Bar(); } } }",
+index: 1);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestUpdateGetSet_GetInSetReference()
+        {
+            Test(
+@"using System; class C { int [||]GetFoo() { } void SetFoo(int i) { } void Bar() { SetFoo(GetFoo() + 1); } }",
+@"using System; class C { int Foo { get { } set { } } void Bar() { Foo = Foo + 1; } }",
+index: 1);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestUpdateGetSet_UpdateSetPrameterName()
+        {
+            Test(
+@"using System; class C { int [||]GetFoo() { } void SetFoo(int i) { v = i; } }",
+@"using System; class C { int Foo { get { } set { v = value; } } }",
+index: 1);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestUpdateGetSet_SetReferenceInSetter()
+        {
+            Test(
+@"using System; class C { int [||]GetFoo() { } void SetFoo(int i) { SetFoo(i - 1); } }",
+@"using System; class C { int Foo { get { } set { Foo = value - 1; } } }",
+index: 1);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestVirtualGetWithOverride_1()
+        {
+            Test(
+@"class C { protected virtual int [||]GetFoo() { } } class D : C { protected override int GetFoo() { } }",
+@"class C { protected virtual int Foo { get { } } } class D : C { protected override int Foo{ get { } } }",
+index: 0);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestVirtualGetWithOverride_2()
+        {
+            Test(
+@"class C { protected virtual int [||]GetFoo() { } } class D : C { protected override int GetFoo() { base.GetFoo(); } }",
+@"class C { protected virtual int Foo { get { } } } class D : C { protected override int Foo { get { base.Foo; } } }",
+index: 0);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestGetWithInterface()
+        {
+            Test(
+@"interface I { int [||]GetFoo(); } class C : I { public int GetFoo() { } }",
+@"interface I { int Foo { get; } } class C : I { public int Foo { get { } } }",
+index: 0);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestWithPartialClasses()
+        {
+            Test(
+@"partial class C { int [||]GetFoo() { } } partial class C { void SetFoo(int i) { } }",
+@"partial class C { int Foo { get { } set { } } } partial class C { }",
+index: 1);
+        }
+    }
+}
```

Comment:
test that we generated correctly within If-Def'd code. Think of shared projects with windows/phone heads. If the user is refactoring a method in Phone context, it should be generated within the active context (if-def path thats active) and should not be generated in code thats common to both windows and phone (that could potentially break code). 

I'll leave it to you to decide how important a scenario this is :)